src/component/mxgraph/BpmnGraph.ts

```diff
@@ -163,9 +163,55 @@ export class BpmnGraph extends mxgraph.mxGraph {
     mxgraph.mxEvent.addMouseWheelListener(throttle(this.createMouseWheelZoomListener(false), config.throttleDelay), this.container);
   }
 
+  /**
+   * @internal
+   */
+  enableNavigation(zoomConfig?: ZoomConfiguration): void {
+    const panningHandler = this.panningHandler;
+
+    // Pan configuration
+    panningHandler.addListener(mxgraph.mxEvent.PAN_START, this.getPanningHandler('grab'));
+    panningHandler.addListener(mxgraph.mxEvent.PAN_END, this.getPanningHandler('default'));
+
+    panningHandler.usePopupTrigger = false; // only use the left button to trigger panning
+    // Reimplement the function as we also want to trigger 'panning on cells' (ignoreCell to true) and only on left-click
+    // The mxGraph standard implementation doesn't ignore right click in this case, so do it by ourselves
+    panningHandler.isForcePanningEvent = (me): boolean => mxgraph.mxEvent.isLeftMouseButton(me.getEvent()) || mxgraph.mxEvent.isMultiTouchEvent(me.getEvent());
+    panningHandler.setPinchEnabled(true);
+    this.setPanning(true);
+
+    // Only set on Graph initizalization
+    if (zoomConfig) {
+      // Zoom configuration
+      this.registerMouseWheelZoomListeners(zoomConfig);
+    }
+  }
+  /**
+   * @internal
+   */
+  disableNavigation(): void {
+    const panningHandler = this.panningHandler;
+    this.setPanning(false);
+    // Remove panningHandler Listeners
+    panningHandler.eventListeners = [];
+    // Disable gesture support for zoom
+    panningHandler.setPinchEnabled(false);
+    // Disable panning on touch device
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- prefix parameter name - common practice to acknowledge the fact that some parameter is unused (e.g. in TypeScript compiler)
+    panningHandler.isForcePanningEvent = (_me: mxMouseEvent): boolean => false;
+  }
+  private getPanningHandler(cursor: 'grab' | 'default'): () => void {
+    return (): void => {
+      this.isEnabled() && (this.container.style.cursor = cursor);
+    };
+  }
+
   // Update the currentZoomLevel when performScaling is false, use the currentZoomLevel to set the scale otherwise
   // Initial implementation inspired by https://github.com/algenty/grafana-flowcharting/blob/0.9.0/src/graph_class.ts#L1254
   private manageMouseWheelZoomEvent(up: boolean, evt: MouseEvent, performScaling: boolean): void {
+    if (!this.panningHandler.pinchEnabled) {
+      return;
+    }
```

Comment:
**question:** why do we need it in the new implementation? I have the feeling (I don't have an environment to test while doing this review) that this is to detect if the navigation is disabled (as pinch is only enabled when navigation is enabled). In this case, we could have an internal state (a navigationEnabled property in the class).
However, if we need the check here, it is probably because we still handle the mouse event when the navigation is disabled. A listener is probably still registered with the risk to register another when if the enable navigation API is call again.

Reply:
I replaced the pinchEnabled to a navigationEnabled property but I also see the risk with not removing the listener ... But I can't really work out when and if at all the listeners are really removed by `mxgraph.mxEvent.removeListener()` maybe you have an idea how this could be solved ?

Reply:
I made some changes to the event registration (use some custom function instead of `mxgraph.mxEvent.addMouseWheelListener`) so they can also be removed if not needed anymore.